pkg/smokescreen/smokescreen.go

```diff
@@ -618,24 +656,24 @@ func logProxy(config *Config, pctx *goproxy.ProxyCtx) {
 	logMethod(CanonicalProxyDecision)
 }
 
-func handleConnect(config *Config, pctx *goproxy.ProxyCtx) error {
+func handleConnect(config *Config, pctx *goproxy.ProxyCtx) (string, error) {
 	sctx := pctx.UserData.(*smokescreenContext)
 
 	// Check if requesting role is allowed to talk to remote
-	remoteHost, err := normalizeHost(pctx.Req.Host, pctx.Req.URL.Scheme)
+	remoteHost, remotePort, err := normalizeHost(pctx.Req.Host, pctx.Req.URL.Scheme, false)
 	if err != nil {
-		pctx.Error = err
-		return pctx.Error
+		pctx.Error = denyError{err}
+		return "", pctx.Error
 	}
-	sctx.decision, sctx.lookupTime, pctx.Error = checkIfRequestShouldBeProxied(config, pctx.Req, remoteHost)
+	sctx.decision, sctx.lookupTime, pctx.Error = checkIfRequestShouldBeProxied(config, pctx.Req, remoteHost, remotePort)
 	if pctx.Error != nil {
-		return pctx.Error
+		return "", pctx.Error
```

Comment:
should this one also be wrapped?

```suggestion
		return "", denyError{pctx.Error}
```

Reply:
It probably should. It wasn't so I didn't touch it. ;)